src/itdelatrisu/opsu/translation/I18n.java

```diff
@@ -0,0 +1,346 @@
+	 * their file name if the key is not present.
+	 * </p>
+	 * @param identifier The name of the language (e.g. English US, English UK,
+	 * Japanese...)
+	 */
+	public static void setLocale(String identifier) {
+		if (identifier == null || identifier.isEmpty())
+			return;
+		if (localeIDs == null)
+			load();
+
+		for (int a = 0; a < localeIDs.length; a++)
+			if (localeIDs[a].equals(identifier)) {
+				currentLocale = LOCALES.get(a);
+				break;
+			}
+	}
+
+	/**
+	 * <p>
+	 * Set the current locale based on the given index. If the
+	 * index is out-of-bounds from the translations, this method does nothing
+	 * and returns.
+	 * </p><p>
+	 * <b>Note:</b> Using this method is discouraged as the indices may change
+	 * per call to {@link #load()}. This method is public to allow options
+	 * to load the localisation settings.
+	 * </p>
+	 * @param index The index that denotes the position of the locale in the
+	 * options
+	 */
+	public static void setLocale(int index) {
+		if (index < 0 || index >= LOCALES.size())
+			return;
+		if(LOCALES.isEmpty())
+			load();
+
+		currentLocale = LOCALES.get(index);
+	}
+
+	/**
+	 * Translates a key to the current language
+	 * @param key The ID of the translation string
+	 * @return A translation of the given key or the key itself when a
+	 * translation does not exist.
+	 */
+	public static String translate(String key) {
+		return getCurrentLocale().translateImpl(key);
+	}
+
+	/**
+	 * Translates a key to the current language and applies formatting via
+	 * {@link String#format(String, Object...)}
+	 * @param key The ID of the translation string
+	 * @param format Arguments to use in formatting. If there are more arguments
+	 * than format specifiers, the extra arguments are ignored.
+	 * @return A translation of the given key as formatted by the given objects
+	 * or the key itself when a translation does not exist.
+	 */
+	public static String translateFormatted(String key, Object... format) {
+		String s = translate(key);
+
+		if (s.equals(key)) // no translation found
+			return key;
+
+		try {
+			return String.format(s, format);
+		} catch (IllegalFormatException var5) {
+			// TODO 'silently fail?'
+			return key;
+		}
+	}
+
+	/**
+	 * Does the actual translation using the given key
+	 * @param key The ID of the translation string
+	 * @return A translation of the given key or the key itself when a
+	 * translation does not exist
+	 */
+	private String translateImpl(String key) {
+		if (key == null)
+			return "null";
+
+		String s = this.keys.get(key);
+
+		// Retry using fallback if no translation key is found
+		if (s == null) {
+			if (!this.equals(FALLBACK))
+				return FALLBACK.translateImpl(key);
+			return key;
+		}
+
+		// process new lines correctly
+		String newlines[] = s.split("\\\\r\\\\n|\\\\n|\\\\r");
+		s = "";
+		for (int a = 0; a < newlines.length; a++)
+			s = s + newlines[a] + (!(a + 1 == newlines.length) ? "\n" : "");
```

Comment:
Unless `s` was an empty string to begin with, you will always add a newline at the end, so you should append it after the loop instead of checking for the last element.

`s = s + ...` can be simplified to `s += ...`.

For the record, `!(a + 1 == newlines.length)` should have been written as `(a + 1 != newlines.length)`, but you won't need the check if you append after the loop.